econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -0,0 +1,1056 @@
+            f"but got shape {np.shape(X)} instead of (n, {self._d_x})"
+        )
+
+        (numeric_index,) = _get_column_indices(X, [feature_index])
+        results = [res for res in self._results
+                   if res.feature_index == numeric_index]
+
+        assert len(results) != 0, f"The feature index supplied was not fitted"
+        (result,) = results
+        return result
+
+    def whatif(self, X, Xnew, feature_index, y):
+        """
+        Get counterfactual predictions when feature_index is changed to Xnew from its observational counterpart.
+
+        Note that this only applies to regression use cases; for classification what-if analysis is not supported.
+
+        Parameters
+        ----------
+        X: array-like
+            Features
+        Xnew: array-like
+            New values of a single column of X
+        feature_index: int or string
+            The index of the feature being varied to Xnew, either as a numeric index or
+            the string name if the input is a dataframe
+        y: array-like
+            Observed labels or outcome of a predictive model for baseline y values
+        """
+
+        assert not self.classification, "What-if analysis cannot be applied to classification tasks"
+
+        assert np.shape(X)[0] == np.shape(Xnew)[0] == np.shape(y)[0], (
+            "X, Xnew, and y must have the same length, but have shapes "
+            f"{np.shape(X)}, {np.shape(Xnew)}, and {np.shape(y)}"
+        )
+
+        assert np.size(feature_index) == 1, f"Only one feature index may be changed, but got {np.size(feature_index)}"
+
+        T0 = _safe_indexing(X, feature_index, axis=1)
+        T1 = Xnew
+        result = self._check_feature_index(X, feature_index)
+        inf = result.estimator.effect_inference(
+            X=result.X_transformer.transform(X), T0=T0, T1=T1)
+
+        # we want to offset the inference object by the baseline estimate of y
+        inf.translate(y)
+
+        return inf
+
+    def _whatif_dict(self, X, Xnew, feature_index, y):
+        """
+        Get counterfactual predictions when feature_index is changed to Xnew from its observational counterpart.
+
+        Note that this only applies to regression use cases; for classification what-if analysis is not supported.
+
+        Parameters
+        ----------
+        X: array-like
+            Features
+        Xnew: array-like
+            New values of a single column of X
+        feature_index: int or string
+            The index of the feature being varied to Xnew, either as a numeric index or
+            the string name if the input is a dataframe
+        y: array-like
+            Observed labels or outcome of a predictive model for baseline y values
+
+        Returns
+        -------
+        dict : dict
+            The counterfactual predictions, as a dictionary
+        """
+
+        inf = self.whatif(X, Xnew, feature_index, y)
```

Comment:
use alphs

Reply:
It turns out not to make sense to add alpha to the regular `whatif` method because the output is just an InferenceResults object that can use any alpha when getting the pvalues, etc. and does not have an alpha baked in.

However, in this method we need to commit to one because the dictionary itself contains the confidence intervals based on a specific confidence level.